platform/state/platform_state.proto

```diff
@@ -0,0 +1,232 @@
+     * The first round to handle transactions SHALL be round `1`.
+     */
+    uint64 round = 1;
+
+    /**
+     * A consensus timestamp for this state.
+     * <p>
+     * This SHALL be the consensus timestamp of the first transaction in
+     * the current round.
+     */
+    proto.Timestamp consensus_timestamp = 2;
+
+    /**
+     * A version describing the current version of application software.
+     * <p>
+     * This SHALL be the software version that created this state.
+     */
+    SoftwareVersion creation_software_version = 3;
+
+    /**
+     * A number of non-ancient rounds.
+     * <p>
+     * This SHALL be the count of rounds considered non-ancient.
+     */
+    uint32 rounds_non_ancient = 4;
+
+    /**
+     * A snapshot of the consensus state at the end of the round.
+     * <p>
+     * This SHALL be used for restart/reconnect.
+     */
+    ConsensusSnapshot consensus_snapshot = 5;
+
+    /**
+     * A timestamp for the next scheduled time when a freeze will start.
+     * <p>
+     * If a freeze is not scheduled, this SHALL NOT be set.<br/>
+     * If a freeze is currently scheduled, this MUST be set, and MUST
+     * match the timestamp requested for that freeze.
+     */
+    proto.Timestamp freeze_time = 6;
+
+    /**
+     * A timestamp for the last time a freeze was performed.<br/>
+     * If not set, there has never been a freeze.
+     */
+    proto.Timestamp last_frozen_time = 7;
+
+    /**
+     * A consensus node software version.<br/>
+     * The software version that enabled birth round mode.
+     * <p>
+     * This SHALL be unset if birth round migration has not yet happened.<br/>
+     * If birth round migration is complete, this SHALL be the _first_ software
+     * version that enabled birth round mode.
+     */
+    SoftwareVersion first_version_in_birth_round_mode = 8;
+
+    /**
+     * A consensus round.<br/>
+     * The last round before the birth round mode was enabled.
+     * <p>
+     * This SHALL be `MAX_UNSIGNED` if birth round mode has not yet been enabled.
+     */
+    uint64 last_round_before_birth_round_mode = 9;
+
+    // Fields below are to be deprecated in the foreseeable future.
+
+    /**
+     * A running event hash.<br/>
+     * This is computed by the consensus event stream.
+     * <p>
+     * This will be _deprecated_ once the consensus event stream is retired.
+     */
+     bytes legacy_running_event_hash = 10000 [deprecated = true];
+
+    /**
+     * A consensus generation.<br/>
+     * The lowest judge generation before birth round mode was enabled.
+     * <p>
+     * This SHALL be `MAX_UNSIGNED` if birth round mode has not yet been enabled.
+     */
+    uint64 lowest_judge_generation_before_birth_round_mode = 10001 [deprecated = true];
+}
+
+
+/**
+ * A consensus node software version.
+ *
+ * This message records version information for configuration, the Hedera
+ * API, and the "Services" subsystems.
+ */
+message SoftwareVersion {
```

Comment:
This is a Hedera specific SoftwareVersion, so it is not compatible with non-Hedera apps. For the software version inside of an event, we used a single semantic version. I am in the process of migrating the events to use this. Also talked to @mhess-swl about it yesterday, he agrees that a single software version is sufficient. It seems that it makes more sense to use a single semantic version for the state as well.

Reply:
Okay, I will remove the custom `SoftwareVersion` then and use `SemanticVersion` instead